libswift/README.md

```diff
@@ -0,0 +1,182 @@
+# Swift implemented in Swift
+
+_Libswift_ is the part of the Swift compiler, which is implemented in Swift.
+
+With _libswift_ it is possible to add SIL optimization passes written in Swift. It allows to gradually migrate the SIL optimizer from C++ to Swift.
+
+## Building
+
+_Libswift_ is a static library and it is built as part of the swift compiler using build-script and CMake.
+
+### CMake integration
+
+_Libswift_ is either built with the Swift package manager or with CMake (see `CMakeLists.txt` in the top-level _libswift_ directory). Currently, the build is done with CMake in case of release-builds or if no package manager is installed on the host system.
+
+The CMake build is done with custom commands and not with CMake's integrated swift support because due to a CMake bug it's not possible to build non top-level static swift libraries.
+
+Debug builds are done with the package manager (if installed), because it works better for incremental builds.
+
+### Build modes
+
+There are four build modes for _libswift_, which can be selected with the `--libswift <mode>` build-script option:
+
+* `disable`: the compiler is not built with _libswift_. In this early stage of development this mode is the default. Right now _libswift_ does not contain any optimizations or features which are not available in the existing optimizer. Therefore a compiler with disabled _libswift_ still behaves as a compiler with enabled _libswift_. This will change soon.
+* `hosttools`: _libswift_ is built with a pre-installed swift toolchain, using a `swiftc` which is expected to be in the command search path. This mode is the preferred way to build for local development, because it is the fastest way to build. It requires a 5.3 (or newer) swift toolchain to be installed on the host system.
+* `bootstrapping`: The compiler and _libswift_ are built with a two-stage bootstrapping process. This is the preferred mode if no swift toolchain is available on the system or if the build should not depend on any pre-installed toolchain.
+* `bs-hostlibs`: This mode is only available on macOS. It's similar to `bootstrapping`, but links the compiler against the host system swift libraries instead of the built libraries. The build is faster than with `bootstrapping` because only the swiftmodule files of the bootstrapping libraries have to be built.
```

Comment:
Could we consider calling this mode `bootstrapping-with-hostlibs`? This would be more clear in the long and would avoid associations with established acronyms in other contexts.

Reply:
ok, done